src/test/groovy/com/mateusz/jakuszko/roomforyou/mapper/ApartmentMapperSpecification.groovy

```diff
@@ -0,0 +1,118 @@
+package com.mateusz.jakuszko.roomforyou.mapper
+
+import com.mateusz.jakuszko.roomforyou.dto.ApartmentDto
+import com.mateusz.jakuszko.roomforyou.entity.Apartment
+import com.mateusz.jakuszko.roomforyou.entity.Customer
+import com.mateusz.jakuszko.roomforyou.entity.Reservation
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import spock.lang.Specification
+
+@SpringBootTest
+class ApartmentMapperSpecification extends Specification {
+
+    @Autowired
+    private ApartmentMapper apartmentMapper
+
+    private static Apartment createApartment() {
+        return Apartment.builder()
+                .id(1L)
+                .city("Terespol")
+                .street("Kraszewskiego")
+                .streetNumber("26")
+                .apartmentNumber(5)
+                .latitude(123.0)
+                .longitude(321.0)
+                .reservations(Collections.singletonList(Reservation.builder().id(11L).build()))
+                .customer(Customer.builder().id(12L).build())
+                .build()
+    }
+
+    private static ApartmentDto createApartmentDto() {
+        return new ApartmentDto.Builder()
+                .id(1L)
+                .city("Terespol")
+                .street("Kraszewskiego")
+                .streetNumber("26")
+                .apartmentNumber(5)
+                .latitude(123.0)
+                .longitude(321.0)
+                .customerId(12L)
+                .reservationsIds(Collections.singletonList(11L))
+                .build();
+    }
+
+    def "assert bean creation"() {
+        expect:
+        apartmentMapper != null
+    }
+
+    def "aap Apartment to ApartmentDto class"() {
```

Comment:
Jak już idziemy w BDD to idzmy na całego 😉 
Gdybyśmy nie implementowali tego przypadku jeszcze w formie "testu w Spock'u" to na kartce można rozpisać go tak:

TestCase: "should convert Apartament to ApartamentDto"
Given: "apartament exists"
When: "apartament is converted by apartment mapper"
Then: "all fields are filled properly" 

Mając taki szkic możemy przejść do implemntacji i zastosować 2 strategie:
- przepisać powyższe tytuły przy znacznikach given/when/then (podobnie jak Ty to zrobiłeś)
- albo co moim zdaniem jeszcze lepsze 😉  opakować kod z każdej sekcji w metody o takich nazwach. Tutaj oczywiście trzeba zachować umiar i niekoniecznie opakowywać jednolinijkowce w dodatkowe metody... ale linie 59-68 spokojnie mogą zostać ukryte w dodatkowej prywatnej metodzie z asercjami. Dzięki temu przy okazji dowiesz się ze muszą być typu void i że trzeba stosować `assert`. Szczegóły w docs'ach

Reply:
Super, wielkie dzięki !!

Reply:
I ogólnie szacun za takie code review :D